src/main/java/de/qabel/desktop/DesktopClientGui.java

```diff
@@ -138,6 +141,19 @@ public void run() {
         );
     }
 
+    @NotNull
+    private QabelTray getQabelTray() throws ClassNotFoundException, UnsupportedLookAndFeelException, InstantiationException, IllegalAccessException {
+        QabelTray tray = new QabelTray(primaryStage, new AwtToast());
+        tray.install();
+        Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
this has nothing to do with the tray icon => the extract-method refactoring was misleading here and you should probably just revert/inline it.